apps/kdf.c

```diff
@@ -100,25 +86,48 @@ int kdf_main(int argc, char **argv)
         goto opthelp;
     }
 
-    id = OBJ_sn2nid(argv[0]);
-    if (id == NID_undef) {
+    if ((kdf = EVP_KDF_fetch(NULL, argv[0], NULL)) == NULL) {
         BIO_printf(bio_err, "Invalid KDF name %s\n", argv[0]);
         goto opthelp;
     }
 
-    ctx = EVP_KDF_CTX_new_id(id);
+    ctx = EVP_KDF_CTX_new(kdf);
     if (ctx == NULL)
         goto err;
 
     if (opts != NULL) {
+        def = EVP_KDF_CTX_settable_params(kdf);
         for (i = 0; i < sk_OPENSSL_STRING_num(opts); i++) {
             char *opt = sk_OPENSSL_STRING_value(opts, i);
-            if (kdf_ctrl_string(ctx, opt) <= 0) {
+            char *stmp, *vtmp;
+
+            stmp = OPENSSL_strdup(opt);
```

Comment:
I thought I changed the mac one to use a app util function for this?

Reply:
You did.  If you merge #9634, it will be available for everyone too :wink:

Reply:
:) Merged

Reply:
The apps are in a sorry state at present :(